packages/util/src/account.ts

```diff
@@ -244,14 +239,16 @@ export const isValidPublic = function (publicKey: Buffer, sanitize: boolean = fa
   assertIsBuffer(publicKey)
   if (publicKey.length === 64) {
     // Convert to SEC1 for secp256k1
-    return publicKeyVerify(Buffer.concat([Buffer.from([4]), publicKey]))
+    // Automatically checks whether point is on curve
+    Point.fromHex(Buffer.concat([Buffer.from([4]), publicKey]));
+    return true;
```

Comment:
will this throw if invalid? should we think about returning false instead, to match isValidPrivate now only returning boolean?

Reply:
Yes, having the same question on review. 🙂